lib/lib.c

```diff
@@ -36,6 +36,17 @@ void __noreturn halt(void) {
     }
 }
 
+/* Trigger Triple Fault */
+void __noreturn hard_reboot(void) {
+    idt_ptr_t idt_ptr = {0};
+
+    cli();
+    lidt(&idt_ptr);
+    int3();
+
+    halt();
```

Comment:
interesting. so does the int3 do anything here?  or otherwise how is the halt() ever reached?

Reply:
The`int3()` trigger the interrupt, which triggers Triple Fault. The `halt()` should not be reached, but it is there as a "precaution".